astropy/nddata/utils.py

```diff
@@ -69,11 +144,46 @@ def extract_array(array_large, shape, position):
     ----------
     array_large : `~numpy.ndarray`
         Array to extract another array from.
-    shape : tuple
-        Shape of the extracted array.
-    position : tuple
-        Position of the small array's center, with respect to the large array.
-        Coordinates should be in the same order as the array shape.
+    shape : tuple or int
+        Shape of the extracted array (for 1D arrays, this can be an int).
+    position : tuple of numbers or number
+         Position of the small array's center, with respect to the large array.
+         Coordinates should be in the same order as the array shape.
+        Integer positions are at the pixel centers. (For 1D arrays, this can be
```

Comment:
@hamogu -- if I had to guess, the sphinx warning is coming from here (or, really, from the two lines above it), because the indentation changes in this section.